device_backends/LogicalNameMapping/src/LNMBitRangeAccessPlugin.cc

```diff
@@ -137,21 +133,22 @@ namespace ChimeraTK::LNMBackend {
             "Register \"" + TransferElement::getName() + "\" with BitRange plugin is not writeable.");
       }
 
-      uint64_t value{};
-      if constexpr(!std::is_integral<UserType>::value) {
-        // This should never be reached, the decorate function should make sure that we only have integral types here
-        assert(false);
-      }
-      else {
-        memcpy(&value, buffer_2D[0].data(), sizeof(UserType));
-      }
+      auto value = fixedPointConverter.toRaw(buffer_2D[0][0]);
+
+      // FIXME: This needs a change in the fixedpoint converter to tell us that it has clamped the value
 
-      // We have received more data than we actually have bits for
-      if((value & ~_baseBitMask) != 0) {
-        this->_dataValidity = DataValidity::faulty;
-        value = _baseBitMask;
+      if constexpr(!std::is_same_v<UserType, Void>) {
+        auto cooked = fixedPointConverter.scalarToCooked<UserType>(value);
+        // NOLINTNEXTLINE(float-equal)
+        if(buffer_2D[0][0] != cooked) {
```

Comment:
I don't think this is working as intended. If the user buffer `buffer_2D[0][0]` contains a floating point value which is not exactly representable by the fixed-point representation (I mean without rounding), the conversion back to the cooked value will result in a different value. In that case we shall not mark the data validity as faulty.

Even if there were no (fixed-point) rounding effects: Computing two times the "same" floating-point value in different ways may not result in two _equal_ floating-point values. This is exactly what the linter warning is about which you are switching off here.

I am not sure how to fix this situation here. My feeling is you won't get away without implementing this properly in the fixed point converter, because that's the only place where you can know reliably if clamping has happened.

On a side note: you can probably get away for now without marking the data as faulty in particular in the write case, since the data validity is discarded in most backends anyway (at least for all numeric addressed backends).

Reply:
Removed the code and replaced it with a comment and a reference to the ticket